lib/spack/spack/test/llnl/util/file_list.py

```diff
@@ -75,12 +80,12 @@ def test_joined_and_str(self, library_list):
         s1 = library_list.joined()
         expected = " ".join(
             [
-                "/dir1/liblapack.%s" % plat_static_ext,
-                "/dir2/libpython3.6.%s"
-                % (plat_apple_shared_ext if sys.platform != "win32" else "dll"),
-                "/dir1/libblas.%s" % plat_static_ext,
-                "/dir3/libz.%s" % plat_shared_ext,
-                "libmpi.%s.20.10.1" % plat_shared_ext,
+                str(Path("/dir1/liblapack.%s" % plat_static_ext)),
+                str(Path("/dir2/libpython3.6.%s"
+                % (plat_apple_shared_ext if sys.platform != "win32" else "dll"))),
+                str(Path("/dir1/libblas.%s" % plat_static_ext)),
+                str(Path("/dir3/libz.%s" % plat_shared_ext)),
+                str(Path("libmpi.%s.20.10.1" % plat_shared_ext)),
```

Comment:
Do these changes mean that every `package.py` also has to modify their code to use `Path` objects too? What does `str(Path(...))` do here--does it normalize the path somehow? Is there a way we can reduce the boilerplate here?

Reply:
@johnwparent has confirmed that package recipes will not be manipulating `Path` objects, but I would still be interested in what `str(Path(...))` is doing here, since it seems to be a no-op.

Reply:
For all the other parameters it's normalizing the path to be platform specific, for the last one it's a regex artifact that is a no-op.

Reply:
Oh my goodness of course! Thanks! I think it would be helpful to explicitly have a function `normalize_string_path()` that calls `str(Path(...))` in `llnl.util.path` or something to make the intent more clear. Does that seem useful?

Reply:
Oh absolutely. I think I mentioned this in the description but if not, these changes were achieved almost purely using regex's so there definitely is some cleaning up/interface improvements that are warranted.

Reply:
Yes sorry I noticed that part too late! Regex replace is a perfectly legitimate refactoring method, I use it all the time via my own emacs plugin https://github.com/cosmicexplorer/helm-rg. Thanks for clarifying!